File: app/src/main/java/com/example/soccernews/data/repositories/NewsRepository.java
package com.example.soccernews.data.repositories;

import androidx.lifecycle.LiveData;

import com.example.soccernews.data.model.News;

import java.util.List;

public interface NewsRepository {
    LiveData<List<News>> getFavoriteNews();
    void setFavoriteNews(News news);
    void removeFavoriteNews(News news);
}

File: app/src/main/java/viana/alison/soccernews/domain/repository/NewsRepository.kt
package viana.alison.soccernews.domain.repository

import kotlinx.coroutines.flow.Flow
import viana.alison.soccernews.data.model.News

interface NewsRepository {
    suspend fun getAllNews(): Flow<List<News>>
    fun getFavoriteNews(): Flow<List<News>>
    fun getFavoriteIds(): Flow<List<Int>>
    suspend fun insertFavoriteNews(news: News)
    suspend fun deleteFavoriteNews(news: News)
}